### PR TITLE
add constraint layout sample

### PR DESCRIPTION
**Pull request by Stallinator**

### CODE REVIEW

File: demo/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -167,6 +167,17 @@
             android:exported="false"
             android:theme="@style/DemoMaterial" />
 
+        <!-- Date -->
```
Comment (Justus-Saringer) [inline demo/src/main/AndroidManifest.xml:170]: I think this shouldn't be Date.

File: demo/src/main/java/de/check24/demo/features/constraintlayout/ComposableConstraintLayoutActivity.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                end.linkTo(parent.end)
+                bottom.linkTo(parent.bottom)
+            }
+        )
+    }
+}
+
+@Preview(showBackground = true, device = Devices.NEXUS_6, showSystemUi = true)
+@Composable
+private fun ConstraintLayoutBarrierPreview() {
```
Comment (Justus-Saringer) [inline demo/src/main/java/de/check24/demo/features/constraintlayout/ComposableConstraintLayoutActivity.kt:98]: Why do you have "Barrier" in the name?

**Review by Justus-Saringer — APPROVED** [review]
Looks good.

File: demo/src/main/java/de/check24/demo/features/constraintlayout/ComposableConstraintLayoutActivity.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+@Composable
+private fun ConstraintLayoutExample() {
+    ConstraintLayout(
+        modifier = Modifier
+            .fillMaxSize()
+            //just for better looks
+            .padding(10.dp)
+    ) {
+        val (text1, text2, text3, text4, text5) = createRefs()
```
Comment (Darius3001) [inline demo/src/main/java/de/check24/demo/features/constraintlayout/ComposableConstraintLayoutActivity.kt:50]: where do you use this?

**Pull request merged**